source/sdk/io/native/FileUnix.ooc

```diff
@@ -316,5 +320,17 @@ version (unix || apple) {
 		mkfifo: override func ~withMode (mode: Int32) -> Int {
 			_mkfifo(path as CString, mode as ModeT)
 		}
+
+		createTempFile: static func (pattern, mode: String) -> FStream {
+			result: FStream
+			mask := umask(S_IRUSR | S_IWUSR)
+			fd := mkstemp(pattern)
```

Comment:
We need to specify the access mask explicitly, because:

> In glibc versions 2.06 and earlier, the file is created with
> permissions 0666, that is, read and write for all users.  This old
> behavior may be a security risk (...)